### PR TITLE
Feat/tina init

### PR DESCRIPTION
**Pull request by logan-anderson**
- [x] Installs tinacms, styled-components, tina-graphql-gateway and tina-graphql-gateway-cli
- [x] Creates a `.tina/schema.ts` file with a blog collection
- [x] Creates a `content/blog` folder with one item `helloWorld.md`
- [x] Creates a `page/blog/[slug].ts` file with all of the Tina pieces wrapped up in one file
  -  getStaticPaths - runs the query to provide slug variables
  -  getStaticProps - runs the query for a single blog post
  - A default page component export is wrapped by `TinaWrapper`, like the component we have in the starter.
 - [ ] ~~Bonus: should we export that component from one of our packages instead to make it more concise?~~

### CODE REVIEW

**Comment by jeffsee55** [comment]
> Bonus: should we export that component from one of our packages instead to make it more concise?

Lets hold off on that for this PR, probably need a bit more discussion

File: packages/tina-graphql-gateway-cli/src/cmds/baseCmds.ts [hunk 1]
```diff
@@ -70,6 +72,12 @@ export const baseCmds: Command[] = [
       "Generate a GraphQL query for your site's schema, (and optionally Typescript types)",
     action: (options) => chain([attachSchema, genTypes], options),
   },
+  {
+    command: INIT,
+    description: 'Add Tinacms to an exsisting project',
```
Comment (DirtyF) [inline packages/tina-graphql-gateway-cli/src/cmds/baseCmds.ts:77]:
```suggestion
    description: 'Add TinaCMS to an existing project',
```

File: packages/tina-graphql-gateway-cli/src/cmds/init/index.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      }
+      resolve(stdout ? stdout : stderr)
+    })
+  })
+}
+
+export async function initTina(ctx: any, next: () => void, options) {
+  console.log(
+    successText(
+      'Welcome to tina Init process sit back and relax while we set up tina for you'
```
Comment (DirtyF) [inline packages/tina-graphql-gateway-cli/src/cmds/init/index.ts:36]:
```suggestion
      'Sit back and relax while we set up Tina for you…'
```

File: packages/tina-graphql-gateway-cli/src/cmds/init/index.ts [hunk 3]
```diff
@@ -46,21 +49,66 @@ export async function installDeps(ctx: any, next: () => void, options) {
...
+  }
+
+  next()
+}
+
+export async function successMessage(ctx: any, next: () => void, options) {
+  const baseDir = process.cwd()
+
+  console.log(`
+What just Happened?
```
Comment (DirtyF) [inline packages/tina-graphql-gateway-cli/src/cmds/init/index.ts:86]:
```suggestion
What just Happened?
```

File: packages/tina-graphql-gateway-cli/src/cmds/init/index.ts [hunk 4]
```diff
@@ -101,10 +102,17 @@ What just Happened?
...
+✅ Setup a your first post in ${blogPostPath}
 
 ✅ ${successText(
-    "Setup a page to play around with tina Start your next dev start with 'next dev' and go to http://locahost:3000/blog/post-one.md to check it out"
-  )}
+    'Setup a page basic Tina Page 🎉'
+  )} start your dev server with 'next dev' and go to http://locahost:3000/demo/blog/HelloWorld.md to check it out"
+
+
+For more Information visit our docs and checkout our getting starter guide
```
Comment (DirtyF) [inline packages/tina-graphql-gateway-cli/src/cmds/init/index.ts:112]:
```suggestion
For more information visit our docs and check out our getting started guide
```

File: packages/tina-graphql-gateway-cli/src/cmds/init/index.ts [hunk 5]
```diff
@@ -101,10 +102,17 @@ What just Happened?
...
+
+Dos: https://tina.io/docs/tina-cloud/
```
Comment (DirtyF) [inline packages/tina-graphql-gateway-cli/src/cmds/init/index.ts:114]:
```suggestion
Docs: https://tina.io/docs/tina-cloud/
```

**Comment by jeffsee55** [comment]
@logan-anderson how can we test this? Is it easiest to just merge it and deploy? We could put this under a canary release

**Comment by logan-anderson** [comment]
> @logan-anderson how can we test this? Is it easiest to just merge it and deploy? We could put this under a canary release

@jeffsee55 Yeah lets do that

File: packages/tina-graphql-gateway-cli/src/cmds/init/setup-files/index.ts [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+}
+
+export const query = \`#graphql
+  query BlogPostQuery($relativePath: String!) {
+    getPostsDocument(relativePath: $relativePath) {
+      data {
+        __typename
+        ... on Article_Doc_Data {
+          title
+          author {
```
Comment (jeffsee55) [inline packages/tina-graphql-gateway-cli/src/cmds/init/setup-files/index.ts:142]: To keep it simple should that autogenerated `schema.ts` _not_ included an author? We can just remove that entirely to keep things simple

**Comment by jamespohalloran** [comment]
Just tested this out on `npx create-next-app --example with-typescript next-demo-app`

This is really awesome! I think this will lead to a huge improvement in conversions. 

I took down a few notes as I went through it:
- npx tina init took a few minutes inially. I wonder if there's a few dependencies we can trim out to speed this up. 

- During the following step, I wasn't sure if I was supposed to run `yarn add tinacms styled-components tina-graphql-gateway tina-graphql-gateway-cli` myself. Maybe this is a matter of formatting, or maybe we don't need to show that bolded line at all (if possible).
Screen Shot 2021-06-11 at 1 59 38 PM
Screen Shot 2021-06-11 at 2 02 47 PM
- With the red text “Collection 'posts', path 'content/posts' not found, check that path exists on your filesystem”: it's hard to know if this is an actual error that I need to act on.
- The yellow “please add the following scripts to your package.json” text is hard to see on light-themed terminal
- There's a lot to take in from this initial body of text. I wonder if we should split the Tina Cloud setup into a separate CLII command, or maybe some extra formatting would help.
- “Start your dev server with next dev” -> this doesn’t work, because I’m not running the graphql server. 
Should this be: `Start your dev server with yarn dev or npm run dev`?
- When running `yarn dev` I got this error (see below). I had to remove the author stuff from the page's query, as it's not defined in the schema
Screen Shot 2021-06-11 at 2 10 58 PM
- I'm interested to see how users feel about always being in edit-mode on this page. It might not be a problem! But we may want to have a decidated page of our docs that talks about how this usually works.

Apart from the few actual errors I ran into, I don't think any of this needs to block its initial rollout! Great job!!

**Comment by jeffsee55** [comment]
Thanks for going through this @jamespohalloran - great points

> With the red text “Collection 'posts', path 'content/posts' not found, check that path exists on your filesystem”: it's hard to know if this is an actual error that I need to act on.

@logan-anderson I believe that's just because we let the default behavior step in when there isn't a `schema.ts`, but can we intentionally build it _first_ so we don't get that message since we're going to be building it anyway.

> There's a lot to take in from this initial body of text. I wonder if we should split the Tina Cloud setup into a separate CLII command, or maybe some extra formatting would help.

Yeah that may make sense, perhaps the part about configuring the .env and stuff could be done as another command.

**Comment by DirtyF** [comment]
> There's a lot to take in from this initial body of text

+1 to point to the next CLI command to  make it a bit less overwhelming, as long as we inform the user about the next steps to take and don't leave them hanging.

**Comment by logan-anderson** [comment]
@jamespohalloran I fixed all of those issues thanks for the comments! 

> npx tina init took a few minutes inially. I wonder if there's a few dependencies we can trim out to speed this up.

Moved some obvious ones into dev dependencies like typescript, tsup and some stuff with '@types/' prefix

Ran ` npx depcheck`
And it looks like we can remove a lot more
```
npx: installed 147 in 6.923s
Unused dependencies
* axios
* clear
* cookie-parser
* crypto
* esm
* express-graphql
* figlet
* generator-code
* http-proxy
* inquirer
* isomorphic-unfetch
* jsonwebtoken
* jwks-rsa
* listr
* ms
* open
* ora
* path
* simple-git
* tina-graphql-helpers
* yaml
* yo
```

@jeffsee55 Anything on this list that should not be removed?

**Review by jeffsee55 — APPROVED** [review]

**Pull request merged**